File: app/src/main/java/io/ermdev/classify/di/module/DatabaseModule.java
package io.ermdev.classify.di.module;

import android.app.Application;

import dagger.Module;
import dagger.Provides;
import io.ermdev.classify.data.local.schedule.ScheduleDao;
import io.ermdev.classify.data.local.schedule.ScheduleDatabase;

/**
 * Created by erafaelmanuel on 11/28/2017.
 */

@Module
public class DatabaseModule {

    @Provides
    public ScheduleDatabase providesScheduleDatabase(Application mApplication) {
        return ScheduleDatabase.getInstance(mApplication);
    }

    @Provides
    public ScheduleDao providesScheduleDao(ScheduleDatabase mScheduleDatabase) {
        return mScheduleDatabase.scheduleDao();
    }
}

File: app/src/main/java/io/ermdev/classify/di/module/DatabaseModule.kt
package io.ermdev.classify.di.module

import android.app.Application
import android.arch.persistence.room.Room
import dagger.Module
import dagger.Provides
import io.ermdev.classify.data.local.AppDatabase
import javax.inject.Singleton

/**
 * Created by erafaelmanuel on 1/3/2018.
 */
@Module
class DatabaseModule {

    @Singleton
    @Provides
    fun providesAppDatabase(mApplication: Application): AppDatabase {
        return Room.databaseBuilder(mApplication, AppDatabase::class.java, "classidy.db")
                .allowMainThreadQueries()
                .build()
    }
}